server.spec.js

```diff
@@ -0,0 +1,56 @@
+const request = require('supertest');
+
+const server = require('./server.js');
+
+
+//Tests for / endpoint
+describe ('get request for /', () => {
+    it('returns a 200 status code', async () => {
+        //get access to the server
+        //use super test to run a get request to server
+        const response = await request(server).get('/');
+
+        expect(response.status).toEqual(200);
+    });
+    it('returns {api: "running"}', async () => {
+        const expectedBody = { api: 'running'}
+        const response = await request(server).get('/');
+        
+        expect(response.body).toEqual(expectedBody);
+    });
+});
+
+describe('get request for users', () => {
+    it('returns a list of users', async () => {
+        let response = await request(server)
+        .get('/users')
+        .set('Accept', 'application/json')
+        .expect('Content-Type', /json/)
+        .expect(200)
+    })
+})
+
+describe('users/:id', () => {
+    it('removes item based on specified id', async () => {
+        let id = '5';
+        let response = await request(server)
+            .delete(`/users/${id}`)
+            .send({message: `user ${id} successfully deleted`})
+            .expect(200)
+    })
+})
+
+describe('/greet/:name', () => {
+    it('greets the person by name', async () => {
+        let first = 'Harry';
+        let last = 'Potter';
+        //make a post passing last name into body
+        //first name is a url parameter
+        let response = await request(server)
+            .post(`/greet/${first}`)
+            .send({ last });
+        //verify that the endpoint returns the right object
+        expect(response.body).toEqual({ hello: 'Harry Potter' });
+    })
```

Comment:
Your testing looks pretty good - I think the error is in the server.js. Your post and delete endpoints look incomplete. Like, your post does not tell the server to post/save anything (though the test works!). The delete might work better with findByIdAndRemove instead of just remove.